src/alt.js

```diff
@@ -364,22 +364,24 @@ class Alt {
     let storeInstance
     let key = iden || StoreModel.name || StoreModel.displayName || ''
 
-    if (saveStore && (this.stores[key] || !key)) {
-      /* istanbul ignore else */
-      if (typeof console !== 'undefined') {
-        if (this.stores[key]) {
-          console.warn(new ReferenceError(
-            `A store named ${key} already exists, double check your store ` +
-            `names or pass in your own custom identifier for each store`
-          ))
-        } else {
-          console.warn(new ReferenceError('Store name was not specified'))
-        }
+    /* istanbul ignore else */
+    if (typeof console !== 'undefined') {
+      if (saveStore && !key) {
+        console.warn('Store name was not specified')
       }
 
-      key = uid(this.stores, key)
+      if (saveStore && this.stores[key] && StoreModel !== this.stores[key].StoreModel) {
+        console.warn(
+          `A store named ${key} already exists, double check your store ` +
+          `names or pass in your own custom identifier for each store`
+        )
+      }
     }
 
+    key = this.stores[key] && StoreModel === this.stores[key].StoreModel
+      ? key
+      : uid(this.stores, key)
+
```

Comment:
nit. Once a ternary has to be on multiple lines, does it make more sense to use an if statement?